From 53e868438c3dc090b94b6966dd2eb604296ed6c1 Mon Sep 17 00:00:00 2001
From: "tpopela@redhat.com"
 <tpopela@redhat.com@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Fri, 17 Mar 2017 09:45:55 +0000
Subject: [PATCH] Fix uninitialized public members in NetworkProcess
 https://bugs.webkit.org/show_bug.cgi?id=169598

Reviewed by Alex Christensen.

Found by Coverity scan.

* NetworkProcess/NetworkProcessCreationParameters.h:
* NetworkProcess/cache/NetworkCacheStorage.cpp:

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@214101 268f45cc-cd09-0410-ab3c-d52691b4dbfc

diff --git a/Source/WebKit2/NetworkProcess/NetworkProcessCreationParameters.h b/Source/WebKit2/NetworkProcess/NetworkProcessCreationParameters.h
index aef7e612988..dd4e2ee113d 100644
--- a/Source/WebKit2/NetworkProcess/NetworkProcessCreationParameters.h
+++ b/Source/WebKit2/NetworkProcess/NetworkProcessCreationParameters.h
@@ -49,18 +49,18 @@ struct NetworkProcessCreationParameters {
     void encode(IPC::Encoder&) const;
     static bool decode(IPC::Decoder&, NetworkProcessCreationParameters&);
 
-    bool privateBrowsingEnabled;
-    CacheModel cacheModel;
+    bool privateBrowsingEnabled { false };
+    CacheModel cacheModel { CacheModelDocumentViewer };
     int64_t diskCacheSizeOverride { -1 };
-    bool canHandleHTTPSServerTrustEvaluation;
+    bool canHandleHTTPSServerTrustEvaluation { true };
 
     String diskCacheDirectory;
     SandboxExtension::Handle diskCacheDirectoryExtensionHandle;
 #if ENABLE(NETWORK_CACHE)
-    bool shouldEnableNetworkCache;
-    bool shouldEnableNetworkCacheEfficacyLogging;
+    bool shouldEnableNetworkCache { false };
+    bool shouldEnableNetworkCacheEfficacyLogging { false };
 #if ENABLE(NETWORK_CACHE_SPECULATIVE_REVALIDATION)
-    bool shouldEnableNetworkCacheSpeculativeRevalidation;
+    bool shouldEnableNetworkCacheSpeculativeRevalidation { false };
 #endif
 #endif
 #if PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 101100
@@ -72,7 +72,7 @@ struct NetworkProcessCreationParameters {
     SandboxExtension::Handle parentBundleDirectoryExtensionHandle;
 #endif
     bool shouldSuppressMemoryPressureHandler { false };
-    bool shouldUseTestingNetworkSession;
+    bool shouldUseTestingNetworkSession { false };
     std::chrono::milliseconds loadThrottleLatency { 0ms };
 
     Vector<String> urlSchemesRegisteredForCustomProtocols;
@@ -98,9 +98,9 @@ struct NetworkProcessCreationParameters {
 
 #if USE(SOUP)
     String cookiePersistentStoragePath;
-    uint32_t cookiePersistentStorageType;
-    HTTPCookieAcceptPolicy cookieAcceptPolicy;
-    bool ignoreTLSErrors;
+    uint32_t cookiePersistentStorageType { 0 };
+    HTTPCookieAcceptPolicy cookieAcceptPolicy { HTTPCookieAcceptPolicyAlways };
+    bool ignoreTLSErrors { false };
     Vector<String> languages;
     WebCore::SoupNetworkProxySettings proxySettings;
 #endif
diff --git a/Source/WebKit2/NetworkProcess/cache/NetworkCacheStorage.cpp b/Source/WebKit2/NetworkProcess/cache/NetworkCacheStorage.cpp
index 46e35dbb5b1..6b3cb093787 100644
--- a/Source/WebKit2/NetworkProcess/cache/NetworkCacheStorage.cpp
+++ b/Source/WebKit2/NetworkProcess/cache/NetworkCacheStorage.cpp
@@ -367,13 +367,13 @@ struct RecordMetaData {
     Key key;
     std::chrono::system_clock::time_point timeStamp;
     SHA1::Digest headerHash;
-    uint64_t headerSize;
+    uint64_t headerSize { 0 };
     SHA1::Digest bodyHash;
-    uint64_t bodySize;
-    bool isBodyInline;
+    uint64_t bodySize { 0 };
+    bool isBodyInline { false };
 
     // Not encoded as a field. Header starts immediately after meta data.
-    uint64_t headerOffset;
+    uint64_t headerOffset { 0 };
 };
 
 static bool decodeRecordMetaData(RecordMetaData& metaData, const Data& fileData)
-- 
2.13.5

